utils/data.helpers.js

```diff
@@ -12,7 +12,11 @@ const getViewData = (req, optionalParams = {}) => {
     params.errors = errors
   }
 
-  return { ...params, ...optionalParams }
+  const data = optionalParams.data
+    ? { ...params.data, ...optionalParams.data }
+    : params.data
+
+  return { ...params, ...optionalParams, data }
```

Comment:
This is so data objects inside the params get merged together rather than just overridden